flyingpigeon/processes/wps_spatial_analog.py

```diff
@@ -0,0 +1,264 @@
+from flyingpigeon.utils import archiveextract
+from flyingpigeon.utils import rename_complexinputs
+from flyingpigeon.utils import get_values
+from flyingpigeon.ocgis_module import call
+from shapely.geometry import Point
+
+from pywps import Process
+from pywps import LiteralInput
+from pywps import ComplexInput, ComplexOutput
+from pywps import Format
+from pywps.app.Common import Metadata
+
+from datetime import datetime as dt
+import os
+
+from ocgis import FunctionRegistry, RequestDataset, OcgOperations
+from flyingpigeon.ocgisDissimilarity import Dissimilarity, metrics
+
+FunctionRegistry.append(Dissimilarity)
+
+import logging
+LOGGER = logging.getLogger("PYWPS")
+
+
+class SpatialAnalogProcess(Process):
+    def __init__(self):
+        inputs = [
+            ComplexInput('candidate', 'Candidate netCDF dataset',
+                         abstract='NetCDF files or archive (tar/zip)  '
+                                  'storing the candidate indices. The output will be stored on this grid.',
+                         metadata=[Metadata('Info')],
+                         min_occurs=1,
+                         max_occurs=1000,
+                         supported_formats=[
+                             Format('application/x-netcdf'),
+                             Format('application/x-tar'),
+                             Format('application/zip'),
+                         ]),
+
+            ComplexInput('target', 'Target netCDF dataset',
+                         abstract='NetCDF files or archive (tar/zip).'
+                                  'containing netcdf files storing the '
+                                  'target indices.',
+                         metadata=[Metadata('Info')],
+                         min_occurs=1,
+                         max_occurs=1000,
+                         supported_formats=[
+                             Format('application/x-netcdf'),
+                             Format('application/x-tar'),
+                             Format('application/zip'),
+                         ]),
+
+            LiteralInput('location', 'Target coordinates (lon,lat)',
+                         abstract="Geographical coordinates (lon,lat) of the target location.",
+                         data_type='string',
+                         min_occurs=1,
+                         max_occurs=1,
+                         ),
+
+            LiteralInput('indices', 'Indices',
+                         abstract="One or more climate indices to use for the comparison.",
+                         data_type='string',
+                         min_occurs=1,
+                         max_occurs=5,
+                         ),
+
+            LiteralInput('dist', "Distance",
+                         abstract="Dissimilarity metric comparing distributions.",
+                         data_type='string',
+                         min_occurs=0,
+                         max_occurs=1,
+                         default='kldiv',
```

Comment:
Yes, allowed_values is metric, a reference to dissimilarity.__all__ which is equal to 
['seuclidean', 'nearest_neighbor', 'zech_aslan',
           'kolmogorov_smirnov', 'friedman_rafsky',
           'kldiv']